.gitlab-ci.yml

```diff
@@ -0,0 +1,71 @@
+stages:
+  - test
+  - build
+
+image:                             parity/rust:nightly
+
+variables:
+  CI_SERVER_NAME:                  "GitLab CI"
+  CARGO_HOME:                      "${CI_PROJECT_DIR}/cargo"
+
+
+cache:
+  key:                             "${CI_JOB_NAME}"
+  paths:
+    - target/
+    - cargo/
+
+
+.collect_artifacts:                &collect_artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     7 days
+    paths:
+    - target/release/polkadot
+
+.determine_version: &determine_version |
+  export VERSION="$(sed -r -n '1,/^version/s/^version = "([^"]+)".*$/\1/p' Cargo.toml)"
+  echo "Version" $VERSION
+
+
+before_script:
+  - ./scripts/build.sh
+
+
+#### stage:                        test
+
+test:rust:stable:
+  stage:                           test
+  variables:
+    RUST_TOOLCHAIN: stable
+    TARGET: native
+  script:
+    - time cargo test --all --release
+  only:
+    - triggers
+    - tags
+    - master
+    - schedules
+    - web
+  tags:
+    - rust-stable 
+
+
+#### stage:                        build
+
+build:linux:ubuntu:amd64:
+  stage:                           build
+  variables:
+    CARGO_TARGET:                  x86_64-unknown-linux-gnu
+    TARGET:                        native
+    RUST_TOOLCHAIN:                stable
```

Comment:
I don't think we need `TARGET` and `RUST_TOOLCHAIN` definitions here. 

We required these defs in Travis CI because we were using `./ci/script.sh` for build script and it was using these variables to configure the build. Because apparently we don't use `./ci/script.sh` here we don't need to define these variables.